lib/elixir/lib/keyword.ex

```diff
@@ -954,6 +954,32 @@ defmodule Keyword do
     end
   end
 
+  @doc """
+  Returns the first value for `key` and removes all associated antries in the keyword list,
+  raising if `key` is not present.
+
+  This function behaves like `pop/3`, but raises in cases the `key` is not present in the
+  given `keywords`.
+
+  ## Examples
+
+      iex> Keyword.pop!([a: 1], :a)
+      {1, []}
+      iex> Keyword.pop!([a: 1, a: 2], :a)
+      {1, []}
+      iex> Keyword.pop!([a: 1], :b)
+      ** (KeyError) key :b not found in: [a: 1]
+
+  """
+  @doc since: "1.10.0"
+  @spec pop!(t, key) :: {value, t}
+  def pop!(keywords, key) when is_list(keywords) and is_atom(key) do
+    case fetch(keywords, key) do
```

Comment:
Is this more efficient than using the new `pop_values/2`?

Reply:
Yes. We don't need to build a list of values to pop just to take the first one. I think it's better to use the same implementation as `pop/3` and just raise in case there's no key.